src/regmod/utils.py

```diff
@@ -40,11 +114,31 @@ def __post_init__(self):
 
     @property
     def num_spline_bases(self) -> int:
+        """Number of the spline bases.
+        """
         inner_knots = self.knots[int(self.l_linear):
                                  len(self.knots) - int(self.r_linear)]
         return len(inner_knots) - 2 + self.degree + int(self.include_first_basis)
 
     def create_spline(self, vec: np.ndarray = None) -> XSpline:
+        """Create spline from the given vector.
+
+        Parameters
+        ----------
+        vec : Optional[np.ndarray]
```

Comment:
This syntax looks different than what's above, try to be consistent, e.g.,
`vec : np.ndarray, optional`

Reply:
Or does this syntax have a specific function, like below where you have `List[int]`?

Reply:
Yes! this is from the `typing` module of python, for more detail see,
https://docs.python.org/3/library/typing.html#typing.Optional
But I agree, I mess the typing with the Numpy docstring style, adding `optional` in at the end.